src/codewhisperer/service/transformByQHandler.ts

```diff
@@ -321,59 +408,130 @@ export function getImageAsBase64(filePath: string) {
+        )
+        const logoBase64 = getImageAsBase64(logoAbsolutePath)
+        let plan = `![Transform by Q](${logoBase64}) \n # Code Transformation Plan by Amazon Q \n\n`
+        plan += CodeWhispererConstants.planIntroductionMessage.replace(
+            'JAVA_VERSION_HERE',
+            transformByQState.getSourceJDKVersion()
+        )
+        plan += `\n\nExpected total transformation steps: ${response.transformationPlan.transformationSteps.length}\n\n`
+        plan += CodeWhispererConstants.planDisclaimerMessage
+        for (const step of response.transformationPlan.transformationSteps) {
+            plan += `**${step.name}**\n\n- ${step.description}\n\n\n`
+        }
 
-    return plan
+        return plan
+    } catch (e: any) {
+        const errorMessage = e?.message || 'Error in GetTransformationPlan API call'
+        telemetry.codeTransform_logApiError.emit({
+            codeTransformApiNames: 'GetTransformationPlan',
+            codeTransformSessionId: codeTransformTelemetryState.getSessionId(),
+            codeTransformJobId: jobId,
+            codeTransformApiErrorMessage: errorMessage,
+            codeTransformRequestId: e?.requestId,
+        })
+        // Pass along error to callee function
+        throw new ToolkitError(errorMessage, { cause: e as Error })
+    }
 }
 
 export async function getTransformationSteps(jobId: string) {
-    await sleep(2000) // prevent ThrottlingException
-    const response = await codeWhisperer.codeWhispererClient.codeModernizerGetCodeTransformationPlan({
-        transformationJobId: jobId,
-    })
-    return response.transformationPlan.transformationSteps
+    try {
+        await sleep(2000) // prevent ThrottlingException
+        const apiStartTime = Date.now()
+        const response = await codeWhisperer.codeWhispererClient.codeModernizerGetCodeTransformationPlan({
+            transformationJobId: jobId,
+        })
+        telemetry.codeTransform_logApiLatency.emit({
+            codeTransformApiNames: 'GetTransformationPlan',
+            codeTransformSessionId: codeTransformTelemetryState.getSessionId(),
+            codeTransformJobId: jobId,
+            codeTransformRunTimeLatency: calculateTotalLatency(apiStartTime),
+            codeTransformRequestId: response.$response.requestId,
+        })
+        return response.transformationPlan.transformationSteps
+    } catch (e: any) {
+        const errorMessage = e?.message || 'Error in GetTransformationPlan API call'
+        telemetry.codeTransform_logApiError.emit({
+            codeTransformApiNames: 'GetTransformationPlan',
+            codeTransformSessionId: codeTransformTelemetryState.getSessionId(),
+            codeTransformJobId: jobId,
+            codeTransformApiErrorMessage: errorMessage,
+            codeTransformRequestId: e?.requestId,
+        })
+        // TODO, do we throw error here or ignore if we are polling?
```

Comment:
Considering that you were throwing uncaught beforehand, you may want to do that here in the interim since this may change functionality now that you're not throwing.